### PR TITLE
Upgrade to recommended settings and use swift 4.2

### PR DESCRIPTION
**Pull request by jcavar**
- Update to Xcode 10 recommended settings
- Use Swift 4.2
- Update to latest macOS image

This project doesn't use any 4.2 specifics so there are no functional changes. This will only silence warning when integrating project.

### CODE REVIEW

File: FBSnapshotTestCaseDemo/Podfile.lock [hunk 1]
```diff
@@ -13,8 +13,8 @@ EXTERNAL SOURCES:
...
 
 SPEC CHECKSUMS:
-  iOSSnapshotTestCase: 037b2e86ebc1cea90a48578c364295cd853c24aa
+  iOSSnapshotTestCase: 57355c72afe193439c624a4f3d150502093d0312
 
-PODFILE CHECKSUM: 23fc15db0c3233d57acaa5dbafbc9f5f89f2f477
+PODFILE CHECKSUM: a061ccaab7eb93ce08d3ae9762dd084a5d79c5ca
 
-COCOAPODS: 1.5.3
+COCOAPODS: 1.5.2
```
Comment (alanzeino) [inline FBSnapshotTestCaseDemo/Podfile.lock]: Don't revert the cocoapods version

File: iOSSnapshotTestCase.podspec [hunk 2]
```diff
@@ -22,7 +22,7 @@ Pod::Spec.new do |s|
   s.pod_target_xcconfig = { 'ENABLE_BITCODE' => 'NO' }
   s.user_target_xcconfig = { 'FRAMEWORK_SEARCH_PATHS' => '$(PLATFORM_DIR)/Developer/Library/Frameworks' }
   s.default_subspecs = 'SwiftSupport'
-  s.swift_version = '4.0'
+  s.swift_version = '4.2'
```
Comment (alanzeino) [inline iOSSnapshotTestCase.podspec]: I don't see any changes in SwiftSupport.swift, so what benefit does this provide?

File: FBSnapshotTestCaseDemo/Podfile.lock (hunk 1 above)
Comment (jcavar) [reply FBSnapshotTestCaseDemo/Podfile.lock]: Sorry this was incidental as my local version was 1.5.2

File: iOSSnapshotTestCase.podspec (hunk 2 above)
Comment (jcavar) [reply iOSSnapshotTestCase.podspec]: There are 2 reasons I think:
- Xcode generates warning for targets that don't use latest Swift version
- If we are about to change/add anything in SwiftSupport we want it to be 4.2 compatible

**Comment by alanzeino** [comment]
Sounds good, I'll merge it once it passes CI

**Comment by jcavar** [comment]
Ok CI is passing now, I had to revert to old os image. 

It seems like something is different in configuration and I have trouble testing locally as well on Xcode 10.

I guess we can merge what we have here and then do this separately if we want?

**Comment by jcavar** [comment]
Could you just tag and release new pod version please?

**Pull request merged**